securesystemslib/signer/_aws_signer.py

```diff
@@ -101,65 +104,85 @@ def import_(cls, aws_key_id: str, local_scheme: str) -> Tuple[str, Key]:
         Raises:
             UnsupportedAlgorithmError: If the AWS KMS signing algorithm is
             unsupported.
-            BotoCoreError: Errors from the botocore.exceptions library.
+            BotoCoreError: Errors from the botocore library.
             ClientError: Errors related to AWS KMS client.
         """
         if AWS_IMPORT_ERROR:
             raise UnsupportedLibraryError(AWS_IMPORT_ERROR)
 
-        client = boto3.client("kms")
-        request = client.get_public_key(KeyId=aws_key_id)
-        kms_pubkey = serialization.load_der_public_key(request["PublicKey"])
-
-        public_key_pem = kms_pubkey.public_bytes(
-            encoding=serialization.Encoding.PEM,
-            format=serialization.PublicFormat.SubjectPublicKeyInfo,
-        ).decode("utf-8")
         try:
-            keytype = cls._get_keytype_for_scheme(local_scheme)
-        except KeyError as e:
-            raise exceptions.UnsupportedAlgorithmError(
-                f"{local_scheme} is not a supported signing algorithm"
-            ) from e
+            client = boto3.client("kms")
+            request = client.get_public_key(KeyId=aws_key_id)
+            kms_pubkey = serialization.load_der_public_key(request["PublicKey"])
+            keytype = cls._get_keytype_from_aws_response(request)
+
+            if keytype == "ecdsa":
+                aws_scheme = request["SigningAlgorithms"][0]
+                scheme = cls._get_ecdsa_scheme(aws_scheme)
```

Comment:
It looks like `local_scheme` is ignored now, if the key type is ecdsa. I think we should fail, if the user passes an unsupported scheme.